### ISSUE TITLE
omit lines (and apptracks) where titles are searcher names (lastname#nnn or 'John Doe' etc)

### ISSUE THREAD

**Issue by caver456** [ncssar/sartopo_bg#18]

**Comment by caver456** [comment]
How can we automatically determine what lines to include on generated debrief PDFs?

("lines" in this context includes unfinished apptracks, finished apptracks with class=Shape, lines acquired from Garmin (imported by whatever method), and lines drawn in the web browser)

Initial thoughts from email 3/31/22:

The first one is a head scratcher - how to exclude lines like 'Joe Searcher' which are finished apptracks judging by the fact that each point in coordinates is a four-element list.  What do you think would be the best method to filter these out?  Some options - maybe there are others?

1 - define a list of folder name(s) whose contents should be excluded from the debrief map, something like 'ExcludeFromDebriefMaps' (but shorter) or the existing folder in the training map 'Tracks to location' or something more generic like 'scratch' or 'temp' or such.  Include all features by default, and require that the operator put superfluous junk in one of those folders.  This might be useful for general planning workflow - not just for dmg - and general map viewability since anyone can easily hide that whole folder.
2 - look at the feature's creator; if it's not the same as one of the folks who created one or more assignments, exclude it - this assumes that top level stuff drawn by someone in the trailer should be included on debrief maps regardless of folder, and lines not drawn by someone in the trailer should always be excluded
3 - look at the line title and try to come to some conclusion from that - but that seems dodgy, error prone, subjective, and hard to maintain
4 - [option added after the initial email] just exclude all finished and unfinished apptracks that don't have proper naming

**Comment by caver456** [comment]
Option 4 looked promising, but it turns out that checking for 4 elements in each point is not a valid way to distinguish an apptrack from a line imported from GPS:

Each point of tracks imported from Garmin 60 (using GPSIO) have 3 elements for tracks that were saved on the GPS (lon/lat/elev), and 4 elements for ACTIVE LOG (lon/lat/elev/timestamp).  This is probably the same for more recent GPSes.

geometry.incremental and geometry.size apparently only exist for unfinished apptracks.

So, from the top.  What are some examples of lines that we want to include on every generated debrief PDF, and lines that we want to exclude from every generated PDF?  (This all assumes that associated lines and apptracks, with appropriate names, are already included on their assignment's PDF only.)

- 'trail' or 'south yuba trail': trail relevant to the search: include for all
  - could be a shape with 2-element points if drawn on the web browser
  - could be a shape with 3-element points if taken from a GPS track
  - could be a shape with 4-element points if it's a finished or unfinished apptrack, or taken from a GPS ACTIVE LOG
  - creator could be an IC worker (if drawn in the trailer, or if it's an IC worker's finished or unfinished apptrack)
  - creator could be a searcher (if it's a searcher's finished or unfinished apptrack)
- 'Joe Searcher' or 'Searcher123': Joe Searcher's apptrack from home to IC: exclude for all
  - could be an unfinished apptrack (with 4-element points)
  - could be a shape with 4-element points (finished apptrack)
  - creator should be the searcher

So, basically, neither the number of points nor the creator nor any function of the two would seem to be a reliable indicator of whether the line should be included or excluded.

**Comment by caver456** [comment]
This probably means going back to option 1: exclude features that are in a certain folder, i.e. show-everything-by-default, which is safer anyway.  The worst case is that the feature is never moved to an excluded folder, and will show up on generated debrief maps - not really that big a deal, and, it's even reasonable to expect that moving features to an excluded folder never becomes a priority during the search.

So: will move forward with option 1 - this could always be reinvestigated later.

**Comment by caver456** [comment]
Actually, should probably leave this open until the filtering is in place and working.  It's actually a bit nontrivial, so, keeping notes here will help.

I >think< these are the actions to take:

- new folder:
  - if name is in excluded names list, add the folder id to excluded folder id list
- folder name change:
  - as above; also, if name is not in excluded names list, make sure folder id is not in excluded folder id list

- new feature other than folder:
  - if folder id exists and is in excluded folder id list, don't import the feature to the debrief map
  - otherwise, import the feature to the debrief map
- folder id change for feature other than folder:
  - if new folder id is in excluded folder id list, make sure the feature does not exist in the debrief map (delete it if it does exist)
  - otherwise, make sure the feature DOES exist in the debrief map (add it if it does not already exist)

**Comment by caver456** [comment]
Note, there's no specific need to remove the id of an excluded folder from the excluded folder id list when that folder is deleted.  It is cleaner to remove it from the list, so if it's easy to do we should do so, but, it won't break anything if the old id remains in the excluded folder id list.

**Comment by caver456** [comment]
Also note that 'folder id change for feature other than folder' needs to happen if the feature has been moved to the default folder, in which case the new properties dict will not have any folderId key, but the previous one did.

**Comment by caver456** [comment]
Also needed to account for moving a source map feature from one excluded folder to another, and the fact that the argument to propertyEditCallback will probably not have a geometry key, since that's the way the engine responds to since requests.

Fixed in commit to plans_console.
